modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +963,284 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
         CV_Error(cv::Error::StsBadArg, "unsupported number of channels");
 }
 
+#if CV_SIMD
+CV_ALWAYS_INLINE void absdiffc_short_store_c1c2c4(short* out_ptr, const v_int32& c1, const v_int32& c2)
+{
+    vx_store(out_ptr, v_pack(c1, c2));
+}
+
+CV_ALWAYS_INLINE void absdiffc_short_store_c1c2c4(ushort* out_ptr, const v_int32& c1, const v_int32& c2)
+{
+    vx_store(out_ptr, v_pack_u(c1, c2));
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c1c2c4(const T in[], T out[],
+                                          const v_float32& s, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint16::nlanes);
```

Comment:
The point is that this function handles cases when data is of unsigned short type and when data is of signed short type. in both cases nlanes is one and the same.

Reply:
So, we are expecting `T=ushort` or `T=short` here; in this case, maybe it would be better to explicitly check that by asserts, smth like:
```
bool isShort = std::is_same<T, ushort>::value || std::is_same<T, short>::value;
GAPI_Assert(isShort == true);
```
This also should be applied to `absdiffc_simd_c3_impl`, I think. There is the same issue

Reply:
> Don't use assumptions in generic implementation (especially silently).

Changed.